File: Factorial.cs
using System;

class FactorialFinding
{
     
    static long Factorial(int n)
    {
        if (n == 0 || n == 1)
            return 1;
        return n * Factorial(n - 1);
    }
	
    static void DisplayResult(int number, long result)
    {
        Console.WriteLine("The factorial of  " +number + " is " +result);
    }

  static void Main(string [] args)
 {
        Console.Write("Enter a number to calculate its factorial: ");
		int number=Convert.ToInt32(Console.ReadLine());
		if(number<0)
		{
		 Console.WriteLine("please enter a valid input:");
		 }
        long result = Factorial(number);
        DisplayResult(number, result);
       
 }
 }

File: Loop/Factorial.js
// Get the command-line argument (number)
const number = parseInt(process.argv[2]);

// Validate input
if (isNaN(number) || number < 0) {
    console.error("Please provide a valid non-negative integer.");
    process.exit(1); // Exit if input is invalid
}

// Function to calculate factorial
function factorial(n) {
    let result = 1;
    for (let i = 1; i <= n; i++) {
        result *= i; // Multiply result by each number up to n
    }
    return result;
}

// Calculate and output the factorial
console.log(`${number}! = ${factorial(number)}`);
